include/ddc/kernels/splines/greville_interpolation_points.hpp

```diff
@@ -172,13 +179,13 @@ class GrevilleInterpolationPoints
             int const n_start
                     = (BcXmin == ddc::BoundCond::GREVILLE) ? BSplines::degree() / 2 + 1 : 1;
             int const domain_size = n_break_points - 2;
+            ddc::DiscreteElement<IntermediateSampling> domain_start(1);
             ddc::DiscreteDomain<IntermediateSampling> const
-                    domain(ddc::DiscreteElement<IntermediateSampling>(1),
-                           ddc::DiscreteVector<IntermediateSampling>(domain_size));
+                    domain(domain_start, ddc::DiscreteVector<IntermediateSampling>(domain_size));
 
             // Copy central points
             ddc::for_each(domain, [&](auto ip) {
-                points_with_bcs[ip.uid() + n_start - 1] = points_wo_bcs.coordinate(ip);
+                points_with_bcs[ip - domain_start + n_start] = points_wo_bcs.coordinate(ip);
```

Comment:
I agree that we remove the call to `uid()` but unfortunately at the price of creating a discrete element from an integer. But I don't see a better way for now.